sys/net/link_layer/nomac/nomac.c

```diff
@@ -0,0 +1,295 @@
+/*
+ * Copyright (C) 2014 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+#include <errno.h>
+#include <string.h>
+
+#include "netapi.h"
+#include "msg.h"
+#include "netdev/base.h"
+#include "thread.h"
+
+#include "nomac.h"
+
+#ifdef MODULE_NETDEV_DUMMY
+#include "netdev_dummy.h"
+#endif
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+#define NOMAC_MSG_QUEUE_SIZE    (16)
+
+static struct {
+    kernel_pid_t registrar_pid; /**< Thread recipient is registered to */
```

Comment:
(Just in case someone tries to argue against it: I don't think it's wise to add a `kernel_pid_t` parameter to the callback type, since `netdev` should be as independent as possible from `netapi`)